sources/packages/backend/libs/services/src/disbursement-overaward/disbursement-overaward.service.ts

```diff
@@ -136,16 +143,36 @@ export class DisbursementOverawardService {
   async addManualOveraward(
     awardValueCode: string,
     overawardValue: number,
+    overawardNotes: string,
     studentId: number,
     auditUserId: number,
   ): Promise<DisbursementOveraward> {
-    const overawardManualRecord = new DisbursementOveraward();
-    overawardManualRecord.creator = { id: auditUserId } as User;
-    overawardManualRecord.disbursementValueCode = awardValueCode;
-    overawardManualRecord.overawardValue = overawardValue;
-    overawardManualRecord.originType =
-      DisbursementOverawardOriginType.ManualRecord;
-    overawardManualRecord.student = { id: studentId } as Student;
-    return this.disbursementOverawardRepo.save(overawardManualRecord);
+    let persistedOverawardManualRecord = null;
```

Comment:
I would say that usually we use the return for this situations as below, avoiding creating `persistedOverawardManualRecord`.
```ts
async addManualOveraward(
    awardValueCode: string,
    overawardValue: number,
    overawardNotes: string,
    studentId: number,
    auditUserId: number,
  ): Promise<DisbursementOveraward> {
    return this.dataSource.transaction(async (transactionalEntityManager) => {
      // Create the note for the overaward and associate the note with the student.
      const noteEntity = await this.noteSharedService.createStudentNote(
        studentId,
        NoteType.Overaward,
        overawardNotes,
        auditUserId,
        transactionalEntityManager,
      );

      // Save disbursement overaward record.
      const overawardManualRecord = new DisbursementOveraward();
      overawardManualRecord.creator = { id: auditUserId } as User;
      overawardManualRecord.disbursementValueCode = awardValueCode;
      overawardManualRecord.overawardValue = overawardValue;
      overawardManualRecord.originType =
        DisbursementOverawardOriginType.ManualRecord;
      overawardManualRecord.student = { id: studentId } as Student;
      overawardManualRecord.overawardNotes = noteEntity;
      overawardManualRecord.addedBy = { id: auditUserId } as User;
      overawardManualRecord.addedDate = new Date();
      return transactionalEntityManager
        .getRepository(DisbursementOveraward)
        .save(overawardManualRecord);
    });
  }
```